content/topics/java-specific/streams/_index.md

```diff
@@ -0,0 +1,69 @@
+---
+content_type: topic
+ready: true
+title: Java Streams
+---
+
+Streams are one of the major features introduced in Java 8. Streams are a sequence of elements that can be easily obtained from a `Collection`. Streams should not be confused with File I/O streams which are entirely different.
+
+## The features of Java streams
+
+- A stream is not a data structure. Instead, a stream takes input from the `Collections`, `Arrays`, or `I/O channels`.
+- Streams don’t change the original data structure, they only provide the result as per the pipelined methods.
+- Each intermediate operation is lazily executed and returns a stream as a result, hence various intermediate operations can be pipelined. Terminal operations mark the end of the stream and return the result.
```

Comment:
Should this actually be 2 bullet points? The thing about terminal operations seems to be a separate point.